File: src/handlers_generated.mjs
// WYGENEROWANE Z contracts.json — NIE EDYTUJ RĘCZNIE. Przegeneruj: `make gen-js`.
// `ok(fields)` (koperta {ok:true,...}) dostarcza pakiet connectora — zaimportuj go:
//   import { ok } from './conn.mjs'
const ok = (fields) => ({ ok: true, ...fields });  // domyślny; nadpisz importem

// WYGENEROWANE Z KONTRAKTU v1 — kształt z contracts.json, nie edytuj ręcznie
export function close({ id = "" } = {}) {
  throw new Error("ciało window/command/close");          // uzupełnij logikę, potem:
  return ok({ action: "window-close", reversible: true, snapshot: {}, inverse: {path: "window/command/restore", args: {snapshot: {}}} });
}

// WYGENEROWANE Z KONTRAKTU v1 — kształt z contracts.json, nie edytuj ręcznie
export function restore({ snapshot = null } = {}) {
  throw new Error("ciało window/command/restore");          // uzupełnij logikę, potem:
  return ok({ action: "window-restore", reversible: true, inverse: {path: "window/command/close", args: {id: ""}} });
}

File: src/handlers_generated.go
// WYGENEROWANE Z contracts.json — NIE EDYTUJ RĘCZNIE. Przegeneruj: `make gen-go`.
package handlers

import "fmt"

var _ = fmt.Errorf // fmt używane w szkieletach poniżej

// WYGENEROWANE Z KONTRAKTU v1 — kształt z contracts.json, nie edytuj ręcznie
type CloseIn struct {
	Id string `json:"id"`
}

func Close(in CloseIn) (map[string]any, error) {
	return nil, fmt.Errorf("ciało window/command/close niezaimplementowane")
	// po implementacji zwróć kopertę zgodną z out-schematem kontraktu
}

// WYGENEROWANE Z KONTRAKTU v1 — kształt z contracts.json, nie edytuj ręcznie
type RestoreIn struct {
	Snapshot map[string]any `json:"snapshot"`
}

func Restore(in RestoreIn) (map[string]any, error) {
	return nil, fmt.Errorf("ciało window/command/restore niezaimplementowane")
	// po implementacji zwróć kopertę zgodną z out-schematem kontraktu
}
